File: src/main/java/com/paragon/client/shader/shaders/FluidShader.java
package com.paragon.client.shader.shaders;

import com.paragon.client.shader.Shader;

import static org.lwjgl.opengl.GL20.*;
import static org.lwjgl.opengl.GL40.glUniform1d;

public class FluidShader extends Shader {

    public FluidShader() {
        super("/assets/paragon/glsl/shaders/fluid.frag");
    }

    @Override
    public void setupUniforms() {
        setupUniform("texture");
        setupUniform("time");
    }

    @Override
    public void updateUniforms() {
        glUniform1i(getUniform("texture"), 0);
        glUniform1f(getUniform("time"), (float) getTime());
    }
}

File: src/main/kotlin/com/paragon/util/render/shader/shaders/FluidShader.kt
package com.paragon.util.render.shader.shaders

import com.paragon.util.render.shader.Shader
import org.lwjgl.opengl.GL20.glUniform1f
import org.lwjgl.opengl.GL20.glUniform1i

class FluidShader : Shader("/assets/paragon/glsl/shaders/fluid.frag") {

    override fun setupUniforms() {
        setupUniform("texture")
        setupUniform("time")
    }

    override fun updateUniforms() {
        glUniform1i(getUniform("texture"), 0)
        glUniform1f(getUniform("time"), time.toFloat())
    }

}